packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,702 @@
+# Integrating a custom plugin in the CLI
+
+One of the main features of Code PushUp is the ability to write custom plugins and track your own metrics.
+It enables you to implement nearly any kind of metric you want to track progress with minimum effort.
+In this section we will go through the steps needed to create a custom plugin and integrate it in your project.
+
+## Setup the core config
+
+To start crafting custom plugins you need a minimum `code-pushup.config.(ts|js|mjs)` file maintaining a `plugins` property.
+property.
+All plugins are registered in a core config object and can take potential options to configure its behaviour.
+The following example shows where to register the plugin:
+
+```typescript
+// code-pushup.config.ts
+import { create as myPlugin } from 'my-plugin';
```

Comment:
- I don't think this would work, without a relative path it'd search in `node_modules`.
- It would be nicer to point users to how we structure imports for our official plugins, I think it's neater than having to rename imports. All that is needed is to add `export default create;` in `my-plugin.ts`.

```suggestion
import myPlugin from './my-plugin';
```